src/backend/distributed/commands/utility_hook.c

```diff
@@ -93,20 +93,40 @@
 	"SELECT citus_internal.start_management_transaction('%lu')"
 #define MARK_OBJECT_DISTRIBUTED \
 	"SELECT citus_internal.mark_object_distributed(%d, %s, %d, %s)"
+#define UNMARK_OBJECT_DISTRIBUTED \
+	"SELECT pg_catalog.citus_unmark_object_distributed(%d, %d, %d,%s)"
+
+/* see NonMainDbDistributedStatementInfo for the explanation of these flags */
+typedef enum DistObjectOperation
+{
+	NO_DIST_OBJECT_OPERATION,
+	MARK_DISTRIBUTED_GLOBALLY,
+	UNMARK_DISTRIBUTED_GLOBALLY
```

Comment:
This should be UNMARK_DISTRIBUTED_LOCALLY because we only unmark the object for local node. The other nodes are handled via utility hook as the comment that we added in this PR explains:

> * Similarly when dropping a distributed object, we always have to unmark such
> * objects as "distributed" and our utility hook on remote nodes achieve this
> * via UnmarkNodeWideObjectsDistributed() because the commands that we send to
> * workers are executed via main db. However for the local node, this is not the
> * case as we're not in the main db. For this reason, callers need to provide
> * UNMARK_DISTRIBUTED_LOCALLY to unmark an object for local node as well.

Reply:
See https://github.com/citusdata/citus/pull/7461#discussion_r1500597562